src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Stream.cs

```diff
@@ -571,6 +672,11 @@ protected override void Dispose(bool disposing)
                 public override int Read(Span<byte> destination)
                 {
                     Http2Stream http2Stream = _http2Stream ?? throw new ObjectDisposedException(nameof(Http2ReadStream));
+                    if (http2Stream._abortException != null)
+                    {
+                        ExceptionDispatchInfo.Throw(new IOException(SR.net_http_client_execution_error, http2Stream._abortException));
```

Comment:
ExceptionDispatchInfo.Throw is to be used when throwing an exception that was previously thrown and caught, enabling you to avoid overwriting the previous stack trace information.  This is a brand new exception; you can just use throw.